packages/api/pages/api/languages/[code]/words/[wordId]/note.ts

```diff
@@ -0,0 +1,52 @@
+import { PatchWordTranslatorNoteRequestBody } from '@translation/api-types';
+import * as z from 'zod';
+import createRoute from '../../../../../../shared/Route';
+import { authorize } from '../../../../../../shared/access-control/authorize';
+import { client } from '../../../../../../shared/db';
+
+export default createRoute<{ code: string; wordId: string }>()
+  .patch<PatchWordTranslatorNoteRequestBody, void>({
+    schema: z.object({
+      note: z.string(),
+    }),
+    authorize: authorize((req) => ({
+      action: 'translate',
+      subject: 'Language',
+      subjectId: req.query.code,
+    })),
+    async handler(req, res) {
+      const language = await client.language.findUnique({
+        where: {
+          code: req.query.code,
+        },
+      });
+
+      if (!language || !req.session || !req.session.user) {
```

Comment:
Can we throw an error if the session user isn't defined because that is an error that shouldn't happen. We would want that to return a 500 status code to indicate an error that shouldn't have happened.

Reply:
Wouldn't that be a 401 error?

Reply:
Yes technically, but this code shouldn't be reachable as long as authorize guarantees the presence of a user